frontend/src/core/comments/components/AddComment.js

```diff
@@ -1,97 +1,393 @@
 /* @flow */
 
 import * as React from 'react';
+import ReactDOM from 'react-dom';
 import { Localized } from '@fluent/react';
+import { 
+    Editor, 
+    Transforms, 
+    Range, 
+    createEditor, 
+    Text,
+    Node,
+} from 'slate';
+import { 
+    Slate, 
+    Editable, 
+    ReactEditor, 
+    withReact,
+} from 'slate-react';
+import escapeHtml from 'escape-html'
 
 import './AddComment.css';
 
 import { UserAvatar } from 'core/user'
 
+import type { TextType, MentionType, InitialType } from 'core/api';
 import type { NavigationParams } from 'core/navigation';
+import type { UserState } from 'core/user';
 
 type Props = {|
-    user: string,
     username: string,
     imageURL: string,
     parameters: ?NavigationParams,
     translation?: ?number,
+    users: UserState,
     addComment: (string, ?number) => void,
+    getUsers: () => void,
 |};
 
 
 export default function AddComments(props: Props) {
     const {
-        user,
         username,
         imageURL,
         parameters,
         translation,
+        users,
         addComment,
+        getUsers,
     } = props;
+       
+    const mentionList: any = React.useRef();
+    const [ target, setTarget ] = React.useState();
+    const [ index, setIndex ] = React.useState(0);
+    const [ search, setSearch ] = React.useState('');
+    const [ scrollPosition, setScrollPosition ] = React.useState(0);
+    const renderElement = React.useCallback(props => <Element { ...props } />, []);
+    const editor = React.useMemo(
+        () => withMentions(withReact(createEditor())), []
+    );
+    const initialValue = [{ type: 'paragraph', children: [{ text: '' }] }];
+    const [ value, setValue ] = React.useState(initialValue);
 
-    const commentInput: any = React.useRef();
-    const minRows = 1;
-    const maxRows = 6;
+    // Set focus on Editor
+    React.useEffect(() => {
+        if (!parameters || parameters.project !== 'terminology') {
+            ReactEditor.focus(editor);
+            Transforms.select(editor, Editor.end(editor, []));
+        }
+    }, [editor, parameters])
+    
+    React.useEffect(() => {
+        getUsers();
+    }, [getUsers])
 
-    if (!user) {
-        return null;
-    }
+    const usersList = users.users;
+    const USERS = usersList.map(user => user.name);
+    const chars = USERS.filter(c =>
```

Comment:
I based much of my initial code off of the example given within the Slate editor documentation. This is a carryover from that. 

With that said, it is technically just `chars` as this is what is rendered as the user is typing in the mention name, thus it may only ever be 1-3 chars until they find the person they wanted and make their selection. At that point it is a `mention` 🤷‍♀️ 

Happy to rename it, but not sure what a better name would be given what it is.

Reply:
This `chars` variable actually is a list of user objects, based on `USERS` filtered to only contain the top 5 users whose names begin with `search`. Unless I'm misunderstanding, `search` is the variable that contains a list of characters. 

Intuitively I would call it something like `suggestedUsers`, as it contains the list of users to pick from that are going to be shown in the interface.

Reply:
And there in lies the danger of responding to reviews late in the day coupled with bad variable naming....Thank you for the friendly response.  😄